src/pybind/mgr/cephadm/module.py

```diff
@@ -854,6 +859,8 @@ def _get_connection(self, host):
         """
         Setup a connection for running commands on remote host.
         """
+        if host is None:
+            raise RuntimeError("_get_connection received NoneType host address")
```

Comment:
Maybe can just assert here?

```suggestion
        assert host
```

And also add some type checking?
```
    def _get_connection(self, host: str):
```